AnimatedComponents/src/components/timer/Timer.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { Text, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native'
+import Animated, {
+  Easing,
+  cancelAnimation,
+  useAnimatedProps,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated'
+import Svg, { Circle as SvgCircle } from 'react-native-svg'
+
+import { TimerPreset } from '@constants'
+
+import styles from './timer-styles'
+
+const AnimatedSvgCircle = Animated.createAnimatedComponent(SvgCircle)
+
+interface ITimerProps {
+  /**buttonStyles: is an optional prop to style the buttons of the timer */
+  buttonStyles?: { container?: ViewStyle; text?: TextStyle }
+  /**circularTimerStrokeColor: is an optional prop that indicates the stroke color of circular timer */
+  circularTimerStrokeColor?: string
```

Comment:
dont take two props `linearTimerColor` and `circularTimerStrokeColor` instead use a prop name `strokeColor`